eng/service.proj

```diff
@@ -81,6 +81,25 @@
             SkipNonexistentTargets="true" />
   </Target>
 
+  <Target Name="GetCodeGenProjects">
+    <MSBuild Projects="@(ProjectReference)"
+            Targets="GetCodeGenProjects"
+            BuildInParallel="$(BuildInParallel)"
+            SkipNonexistentProjects="false"
+            SkipNonexistentTargets="true">
+      <Output ItemName="ProjectsToInclude" TaskParameter="TargetOutputs"/>
+    </MSBuild>
+    <RemoveDuplicates Inputs="@(ProjectsToInclude->Replace($(RepoRoot), ''))">
+      <Output TaskParameter="Filtered" ItemName="ProjectsToIncludeFitered"/>
+    </RemoveDuplicates>
+    <Message Text="Write to file $(OutputProjectFilePath)" Importance="high"/>
+    <Message Text="%24%28RepoRoot%29%(ProjectsToIncludeFitered.Identity)" Importance="high"/>
```

Comment:
This isn't supposed to be encoded, is it? Doesn't seem like you'd get the result you want in the output file.

Reply:
The output is correct in the file. We want the lines to start with $(RepoRoot) in the file.   We have to encode it here.

Reply:
Okay, so it gets read in as-is and evaluated later?

Reply:
Right.  The file emits with lines like  "$(RepoRoot)svc/storage/...".   There's powershell in the AutoRest.CSharp repo that separates the lines into groups, then wraps those in ProjectReference items in a project file.    The project file is imported into service.proj where RepoRoot is already defined

Reply:
This mirrors the behavior of the ProjectDependsOn below